tools/matchcompiler.py

```diff
@@ -380,44 +380,46 @@ def _replaceSpecificTokenMatch(
         )
 
     def _replaceTokenMatch(self, line, linenr, filename):
-        while True:
-            is_simplematch = False
-            pos1 = line.find('Token::Match(')
-            if pos1 == -1:
-                is_simplematch = True
-                pos1 = line.find('Token::simpleMatch(')
-            if pos1 == -1:
-                break
-
-            res = self.parseMatch(line, pos1)
-            if res is None:
-                break
-
-            # assert that Token::Match has either 2 or 3 arguments
-            assert(len(res) == 3 or len(res) == 4)
-
-            end_pos = len(res[0])
-            tok = res[1]
-            raw_pattern = res[2]
-            varId = None
-            if len(res) == 4:
-                varId = res[3]
+        for func in ('Match', 'simpleMatch'):
```

Comment:
I was a bit back and forth if this should be a function which is called twice, instead of having a loop, but I guess it doesn't matter that much.